### PR TITLE
main/u-boot: added support for sopine

### PR DESCRIPTION
**Pull request by MartijnBraam**
This adds support for the PINE A64-LTS board and should work on the sopine board
The patch adds the sopine code that isn't upstream yet

### CODE REVIEW

File: main/u-boot/APKBUILD [hunk 1]
```diff
@@ -4,21 +4,25 @@
 # Maintainer: Leonardo Arena <rnalrd@alpinelinux.org>
 pkgname=u-boot
 pkgver=2018.05
-pkgrel=2
+pkgrel=3
+_atfversion="c9f55c023164a6c8c49f70f7ac6c68c626839d6f"
 pkgdesc="u-boot bootloader common files"
 url="http://www.denx.de/wiki/U-Boot/"
-arch="armhf armv7 aarch64"
+arch="armhf aarch64"
```
Comment (z3ntu) [inline main/u-boot/APKBUILD]: This is probably not on purpose?

Comment (MartijnBraam) [reply main/u-boot/APKBUILD]: No it isn't

**Comment by rnalrd** [comment]
Would you pls rebase? Thanks!

**Comment by z3ntu** [comment]
This PR should be closed in favor of updating u-boot to 2019.01 (all required stuff is upstream now). Our current diff in postmarketOS is:
```diff
diff -u -r /tmp/u-boot-alpine/APKBUILD u-boot/APKBUILD
--- /tmp/u-boot-alpine/APKBUILD	2019-02-08 14:34:18.563897134 +0100
+++ u-boot/APKBUILD	2019-02-06 20:19:33.500288979 +0100
@@ -3,8 +3,9 @@
 # Contributor: Timo Teras <timo.teras@iki.fi>
 # Maintainer: Leonardo Arena <rnalrd@alpinelinux.org>
 pkgname=u-boot
-pkgver=2018.05
-pkgrel=5
+pkgver=2019.01
+pkgrel=0
+_atfversion="f0bfe15b81e50c6561a3a9a8e269db0267778910"
 pkgdesc="u-boot bootloader common files"
 url="http://www.denx.de/wiki/U-Boot/"
 arch="armhf armv7 aarch64"
@@ -12,13 +13,15 @@
 	ISC LGPL-2.0-only LGPL-2.1-only X11"
 depends=""
 depends_dev=""
-makedepends="$depends_dev bc dtc python2-dev swig"
+makedepends="$depends_dev bc dtc python2-dev swig binutils coreutils bison flex"
 install=""
-source="ftp://ftp.denx.de/pub/u-boot/u-boot-${pkgver//_/-}.tar.bz2
+source="http://ftp.denx.de/pub/u-boot/u-boot-${pkgver//_/-}.tar.bz2
+	arm-trusted-firmware-${_atfversion}.tar.gz::https://github.com/ARM-software/arm-trusted-firmware/archive/${_atfversion}.tar.gz
 	README.txt
 	update-u-boot
 	"
 builddir="$srcdir"/u-boot-${pkgver//_/-}
+atfbuilddir="$srcdir"/arm-trusted-firmware-${_atfversion}
 
 case "$CARCH" in
 arm*) board_configs="
@@ -32,7 +35,7 @@
 	thunderx:thunderx_88xx
 	raspberrypi:rpi_3
 	odroid:odroid-c2
-	libretech:libretech-cc
+	pine64:pine64-lts
 	";;
 esac
 
@@ -43,6 +46,19 @@
 subpackages="$pkgname-all:_all $_allboards"
 
 build() {
+	if [ "$CARCH" = "aarch64" ]; then
+		export LDFLAGS=""
+		msg "Building ARM trusted firmware for allwinner"
+		cd "$atfbuilddir"
+		make PLAT=sun50i_a64 DEBUG=1 bl31
+		export BL31="$atfbuilddir"/build/sun50i_a64/debug/bl31.bin
+		# Sanity check $BL31
+		if [ ! -f "$BL31" ]; then
+			echo 'The $BL31 variable is invalid. Please adjust.'
+			return 1
+		fi
+	fi
+
 	cd "$builddir"
 	touch include/config.h
 	LC_ALL=C date +'#define U_BOOT_DATE "%b %d %C%y"' > include/timestamp_autogenerated.h
@@ -107,6 +123,7 @@
 	eval "${_board}() { _split_boards $_board ${_configs//,/ }; }"
 done
 
-sha512sums="218f71282c9d027b0faf210ce2d4d713779e0c61ea4a23f03d5dec0ca8836a3c26b46c5e258d583834f5b5a2f4663f98729ca4b0101a60f85457eb892f8370d2  u-boot-2018.05.tar.bz2
+sha512sums="d9699cd22afe9bc747d64208068c2cf8a2c3143d161ede24536f6fd6adfd6b81e28920589722639e2e48fcf34e8dbde3ead7f691f14cbcc38cd75694d14d719b  u-boot-2019.01.tar.bz2
+732b806bf3a612ded9351226c8fae78d02741220034009350be1c2237341584bcb366c0db2a19b64e4548e9a720d9380d1240680459b831943d3936077eecb13  arm-trusted-firmware-f0bfe15b81e50c6561a3a9a8e269db0267778910.tar.gz
 f8c9bb6e84d6f0620c976ac7ad5dd7ec7ff9dfdd4b1d03d2bf6653e7beccf80bdf2debfc92fb1f696dba92fb40287d3c45897e0078951451d0835cb61a5f16d1  README.txt
-b3b4fb70c4eea2034bae45c513bca7b738be7507b46d3f2d62b31c794cd238d492acf960c95078cb1c945520debd5d2762103df7cef0fc36acfe9ff9a94549fc  update-u-boot"
+f3bb656bbd13caf10a8ea619bc08b439e7dc27526c9cbdfa0374f4f8e277632ff4df06b34d441d6723a93d4a54aa03465d4270c44219d6a150d416b9c56edbbe  update-u-boot"
diff -u -r /tmp/u-boot-alpine/update-u-boot u-boot/update-u-boot
--- /tmp/u-boot-alpine/update-u-boot	2019-02-08 14:34:18.847217037 +0100
+++ u-boot/update-u-boot	2019-02-06 20:19:33.500288979 +0100
@@ -10,6 +10,7 @@
 	if [ -z "$board" -a -e /sys/firmware/devicetree/base/compatible ]; then
 		case "$(cat /sys/firmware/devicetree/base/compatible 2>/dev/null)" in
 		wand,*) board=wand ;;
+		pine64,*) board=pine64 ;;
 		esac
 	fi
 
@@ -39,7 +40,7 @@
 
 options:
 
- -b,--board <board>       Specify the board type: wand, cubie, cubie2, cuboxi
+ -b,--board <board>       Specify the board type: wand, cubie, cubie2, cuboxi, pine64
                           (current default: ${board:-none})
 
  -d,--device <device>     Specify the device where to install u-boot
@@ -62,6 +63,7 @@
 		wand|wandboard) board="wand" ;;
 		cubie|cubieboard) board="cubie" ;;
 		cuboxi|mx6cuboxi) board="mx6cuboxi" ;;
+		pine64) board="pine64" ;;
 		*) usage; exit 1;;
 		esac
 		shift
@@ -115,6 +117,10 @@
 	$dryrun dd if=$imagedir/mx6cuboxi
```

The arm-trusted-firmware part should probably be offloaded to a different package but other than that this diff works on the Pine A64-LTS.

**Pull request closed without merging**